plugins/repository-gcs/src/main/java/org/elasticsearch/plugin/repository/gcs/GoogleCloudStorageModule.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.plugin.repository.gcs;
+
+import org.elasticsearch.common.inject.AbstractModule;
+import org.elasticsearch.repositories.gcs.GoogleCloudStorageService;
+import org.elasticsearch.repositories.gcs.InternalGoogleCloudStorageService;
+
+public class GoogleCloudStorageModule extends AbstractModule {
+
+    @Override
+    protected void configure() {
+        bind(GoogleCloudStorageService.class).to(InternalGoogleCloudStorageService.class).asEagerSingleton();
```

Comment:
Since we bind to concrete implementation here, maybe it makes sense to get rid of GoogleCloudStorageService interface all together and rename InternalGoogleCloudStorageService to GoogleCloudStorageService. It feels like these interfaces don't buy us anything except added complexity. Replacing them with mock implementation would make little sense here since we already test all upper levels with mock implementation based on fs repo.